src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  return sourceString
+    .split(';')
+    .map((keyVal) => keyVal.split(':').map((str) => str.trim()))
+    .filter(([key, value]) => key && value)
+    .reduce((accumulator, [key, value]) => {
```

Comment:
The use of `reduce` to build up the final object from the array of key-value pairs is a good approach and follows the checklist's recommendation to avoid loops and use iteration methods instead.